cpp/biconnected_components_module/algorithm/biconnected_components.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <stack>
+
+#include <mg_graph.hpp>
+
+namespace bcc_utility {
+
+/// Simple struct that keeps the state of nodes in algorithms that rely on
+/// DFS traversal.
+struct NodeState {
+  std::vector<bool> visited;
+  std::vector<std::uint64_t> discovery, low_link, depth;
+  std::vector<std::uint64_t> parent;
+  uint32_t counter;
+};
+
+///
+///@brief DFS Algorithm for obtaining the biconnected components within the graph
+///
+///@param node_id Starting Node ID
+///@param parent_id Parental Node ID
+///@param state Current node state
+///@param edge_stack Current state of edge stack
+///@param bcc Current biconnected components
+///@param graph Graph to work on
+///
+void BccDFS(std::uint64_t node_id, std::uint64_t parent_id, bcc_utility::NodeState *state,
+            std::stack<mg_graph::Edge<>> *edge_stack, std::vector<std::vector<mg_graph::Edge<>>> *bcc,
+            const mg_graph::GraphView<> *graph);
```

Comment:
const reference for graph?

Reply:
Why shouldn't it be const, nothing within the graph should be changed inside the function?

Reply:
```suggestion
            const mg_graph::GraphView<> &graph);
```
I meant this.